pkg/reconciler/pipelinerun/resources/apply_test.go

```diff
@@ -2034,6 +2034,284 @@ func TestApplyParameters_ArrayIndexing(t *testing.T) {
 	}
 }
 
+func TestApplyReplacementsMatrix(t *testing.T) {
+	ctx := context.Background()
+	cfg := config.FromContextOrDefaults(ctx)
+	cfg.FeatureFlags.EnableAPIFields = config.AlphaAPIFields
+	ctx = config.ToContext(ctx, cfg)
```

Comment:
Not block for this PR. This could be replaced with 
```
ctx:=config.SetEnableAPIFields(ctx, config.AlphaAPIFields)
```
But seems we don't export that function. 
I tried to export it here https://github.com/tektoncd/pipeline/pull/6280/files